src/wazuh_db/wdb_integrity.c

```diff
@@ -57,7 +57,7 @@ extern void mock_assert(const int result, const char* const expression,
 #endif
 
 void wdbi_report_removed(const char* agent_id, wdb_component_t component, sqlite3_stmt* stmt) {
-    if (!router_fim_events_handle || !router_inventory_events_handle) {
+    if (!router_fim_events_handle && !router_inventory_events_handle) {
```

Comment:
Question: why do we change "any of the handles missing is an error" to "only is an error to have both missing"?

Reply:
These two handles were added besides the existent one (router_agent_event_handle (or something like that). I found that a mistake, because the handle is always fim or inventory, I don't think we should get the error message "Router handle not available" when we certainly have the router handle.

Reply:
I've reviewed the code and we have 3 handles:
- WDB_AGENT_EVENTS_TOPIC: is used to report the agent deletion and/or upgrade
- WDB_FIM_EVENTS_TOPIC: reports RSync deletion events for FIM
- WDB_INVENTORY_EVENTS_TOPIC: reports RSync deletion events for syscollector

This method at `wdb_integrity` is in charge of reporting RSync deletion events, and the function `wdbi_report_removed()` will use the corresponding handle according to the component. The WDB_AGENT_EVENTS_TOPIC handle isn't involved here.

Checking if any of the handles isn't available is a way to fail early.
We can improve the logging and say that the one of the handles for deletion events isn't available, but we shouldn't change this behavior in this issue.